### DISCUSSION TITLE
Error: Dynamic Code Evaluation (e. g. 'eval', 'new Function', 'WebAssembly.compile') not allowed in Edge Runtime

### DISCUSSION THREAD

**Discussion by alananisimov** [nextauthjs/next-auth#10959] [Help]
I created custom credentials provider, that works on stellar-sdk. This library is using WebAssembly.compile func, which is not supported by edge runtime. Maybe somehow I can switch to Nodejs runtime.

Code: 
`import slugify from "@sindresorhus/slugify";
import { Keypair, StrKey } from "@stellar/stellar-sdk";
import Credentials from "next-auth/providers/credentials";

import { Register } from "@acme/backend";
import { secretKeyProviderSchema } from "@acme/validators";

export const secretKeyProvider = Credentials({
  name: "secret_key",
  credentials: {
    secret_key: { label: "Secret Key", type: "text" },
    wallet_name: { label: "Wallet name", type: "text" },
    ref_id: { label: "Referral ID", type: "number" },
  },
  async authorize(credentials) {
    if (!credentials) return null;
    const {
      secret_key: sk,
      wallet_name,
      ref_id,
    } = await secretKeyProviderSchema.parseAsync(credentials);
    const valid_sk = StrKey.isValidEd25519SecretSeed(sk);

    if (!valid_sk) return null;
    var sourceKeypair = Keypair.fromSecret(sk);
    var sourcePublicKey = sourceKeypair.publicKey();
    var ciphertext = CryptoJS.AES.encrypt(sk, sourcePublicKey);
    var wallet = {
      pk: sourcePublicKey,
      sk: ciphertext.toString(),
      title: wallet_name ? slugify(wallet_name) : "My Wallet",
    };
    const data = await Register({
      wallet,
      refId: ref_id,
    });
    if (!data) {
      return null;
    }

    return { api: data, wallets: [wallet] };
  },
});
`
Error: Dynamic Code Evaluation (e. g. 'eval', 'new Function', 'WebAssembly.compile') not allowed in Edge Runtime 
Learn More: https://nextjs.org/docs/messages/edge-dynamic-code-evaluation
The error was caused by importing '@stellar/stellar-sdk/dist/stellar-sdk.min.js' in '../../packages/auth/src/providers/secret-key.ts'.
`

**Comment by k-taro56** [comment]
Do you want to use the `Node.js` runtime?
If so, make sure that `export const runtime = "edge"` is **not** in `. /app/api/auth/[...nextauth]/route.ts`. Usually `Route Handlers` work with `Node.js` runtime.
https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#runtime

**Reply by alananisimov** [reply]
That is not working. I removed all `export const runtime = ` from auth files

**Reply by k-taro56** [reply]
Could you show me your repo?

**Reply by k-taro56** [reply]
And this is probably not a `next-auth` issue, so I would suggest you ask elsewhere.

**Reply by AbuSantos** [reply]
hey @k-taro56 , im having the same error,
